<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QMainWindow</class>
 <widget class="QMainWindow" name="QMainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>378</width>
    <height>498</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>LPCProg v0.1 (currently for LPC214X)</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout" columnstretch="0,1">
    <item row="1" column="0">
     <widget class="QLabel" name="crystal_label">
      <property name="text">
       <string>Crystal Value(KHz)</string>
      </property>
      <property name="buddy">
       <cstring>crystal_spinBox</cstring>
      </property>
     </widget>
    </item>
    <item row="9" column="0">
     <widget class="QPushButton" name="save_pushButton">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="toolTip">
       <string>Save chip firmware to file</string>
      </property>
      <property name="text">
       <string>&amp;Save</string>
      </property>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QComboBox" name="ports_comboBox"/>
    </item>
    <item row="6" column="0">
     <widget class="QPushButton" name="erase_pushButton">
      <property name="toolTip">
       <string>Erase chip</string>
      </property>
      <property name="text">
       <string>&amp;Erase</string>
      </property>
     </widget>
    </item>
    <item row="10" column="0">
     <widget class="QPushButton" name="decompile_pushButton">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="toolTip">
       <string>Read and decompile chip firmware</string>
      </property>
      <property name="text">
       <string>Read and &amp;decompile</string>
      </property>
     </widget>
    </item>
    <item row="3" column="1">
     <widget class="QLabel" name="chipID_label"/>
    </item>
    <item row="0" column="0">
     <widget class="QLabel" name="ports_label">
      <property name="text">
       <string>Serial port</string>
      </property>
      <property name="buddy">
       <cstring>ports_comboBox</cstring>
      </property>
     </widget>
    </item>
    <item row="4" column="0">
     <widget class="QPushButton" name="firmwareVersion_pushButton">
      <property name="toolTip">
       <string>Get chip ISP firmware version</string>
      </property>
      <property name="text">
       <string>Get firmware version</string>
      </property>
     </widget>
    </item>
    <item row="4" column="1">
     <widget class="QLabel" name="firmwareVersion_label"/>
    </item>
    <item row="3" column="0">
     <widget class="QPushButton" name="chipID_pushButton">
      <property name="toolTip">
       <string>Get chip ID</string>
      </property>
      <property name="text">
       <string>&amp;Get chip ID</string>
      </property>
     </widget>
    </item>
    <item row="11" column="0" colspan="2">
     <widget class="QListWidget" name="listWidget"/>
    </item>
    <item row="1" column="1">
     <widget class="QSpinBox" name="crystal_spinBox">
      <property name="minimum">
       <number>8000</number>
      </property>
      <property name="maximum">
       <number>20000</number>
      </property>
      <property name="singleStep">
       <number>100</number>
      </property>
      <property name="value">
       <number>12000</number>
      </property>
     </widget>
    </item>
    <item row="8" column="0">
     <widget class="QPushButton" name="read_pushButton">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="toolTip">
       <string>Read chip firmware and display the content</string>
      </property>
      <property name="text">
       <string>&amp;Read and display</string>
      </property>
     </widget>
    </item>
    <item row="2" column="0" colspan="2">
     <widget class="Line" name="line">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item row="5" column="0" colspan="2">
     <widget class="QGroupBox" name="groupBox">
      <property name="title">
       <string>File Operation</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QRadioButton" name="fileProgram_radioButton">
           <property name="toolTip">
            <string>Select to program file to chip</string>
           </property>
           <property name="text">
            <string>&amp;Program</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="fileVerify_radioButton">
           <property name="toolTip">
            <string>Select to verify chip firmware to hex file</string>
           </property>
           <property name="text">
            <string>&amp;Verify</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="fileDecompile_radioButton">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="toolTip">
            <string>Select to decompile hex file</string>
           </property>
           <property name="text">
            <string>&amp;Decompile</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QLineEdit" name="file_lineEdit">
           <property name="toolTip">
            <string>Firmware file</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QToolButton" name="fileBrowse_toolButton">
           <property name="toolTip">
            <string>Browse for firmware</string>
           </property>
           <property name="text">
            <string>...</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <spacer name="fileOperationLeft_horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="fileOperation_pushButton">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="toolTip">
            <string>Run selected command</string>
           </property>
           <property name="text">
            <string>&amp;Go</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="fileOperationRigth_horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item row="7" column="0">
     <widget class="QPushButton" name="blankCheck_pushButton">
      <property name="text">
       <string>&amp;Blank Check</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <tabstops>
  <tabstop>ports_comboBox</tabstop>
  <tabstop>crystal_spinBox</tabstop>
  <tabstop>chipID_pushButton</tabstop>
  <tabstop>firmwareVersion_pushButton</tabstop>
  <tabstop>fileProgram_radioButton</tabstop>
  <tabstop>fileVerify_radioButton</tabstop>
  <tabstop>fileDecompile_radioButton</tabstop>
  <tabstop>file_lineEdit</tabstop>
  <tabstop>fileBrowse_toolButton</tabstop>
  <tabstop>fileOperation_pushButton</tabstop>
  <tabstop>erase_pushButton</tabstop>
  <tabstop>blankCheck_pushButton</tabstop>
  <tabstop>read_pushButton</tabstop>
  <tabstop>save_pushButton</tabstop>
  <tabstop>decompile_pushButton</tabstop>
  <tabstop>listWidget</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
